### PR TITLE
Autreg 72 update cypress tests

### PR DESCRIPTION
**Pull request by Eiriknil**

### CODE REVIEW

File: api/src/main/java/no/bibsys/db/RegistryManager.java [hunk 1]
```diff
@@ -178,14 +179,21 @@ public boolean deleteRegistry(String registryName) {
     }
 
     public void setSchemaJson(String registryName, String schemaAsJson) throws IOException {
+        
+        if (tableDriver.tableSize(registryName) > 0) {
```
Comment (janbj) [inline api/src/main/java/no/bibsys/db/RegistryManager.java]: We are repeating these 3 lines, maybe we should create a service method?

File: cypress.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-	"baseUrl": "https://s7trqv8kbg.execute-api.eu-west-1.amazonaws.com/test",
+	"baseUrl": "https://0wkx9ubycd.execute-api.eu-west-1.amazonaws.com/test",
```
Comment (janbj) [inline cypress.json]: Remove this?

File: cypress/integration/APIadmin.feature [hunk 3]
```diff
@@ -34,22 +34,23 @@ Feature: API admin features
...
-    And that there is an existing, populated entity registry with a schema and registered registry API keys
-    When the API admin user removes registry admin API keys from the entity registry
-    Then the API keys no longer provide access to the entity registry
-
   Scenario: An API admin user updates the entity registry metadata
     And that there is an existing, populated entity registry with a schema
     When the API admin user changes the metadata for the entity registry
     Then the metadata for the entity registry is updated
+
+#  Scenario: An API admin user deletes populated data from an entity registry
```
Comment (janbj) [inline cypress/integration/APIadmin.feature]: Why are the scenarios commented out?

File: cypress/integration/anonymousUser.feature [hunk 4]
```diff
@@ -2,9 +2,9 @@ Feature: Anonymous user features
 
   Background:
     Given that there is an existing entity registry with a schema
-
+ 
   Scenario: An anonymous user views an entity without specifying a format
-    Given that there is an entity in the registry
+    And that there is an entity in the registry
```
Comment (janbj) [inline cypress/integration/anonymousUser.feature]: And is in addition to the previous Given?

File: cypress/support/commands.js [hunk 5]
```diff
@@ -23,3 +23,122 @@
...
+})
+
+Cypress.Commands.add("createEntity", (registryName, apiKey, dataFile) => {
+	createEntity(registryName, apiKey, dataFile);
+})
+
+function waitUntilRegistryIsReady(registryName, count){
+
+	let statusUrl = '/registry/' + registryName + '/status'
+	cy.log('waiting...')
```
Comment (janbj) [inline cypress/support/commands.js]: log waiting for registry to become ready?

File: cypress/support/commands.js [hunk 6]
```diff
@@ -23,3 +23,122 @@
...
+	cy.request({
+		url: statusUrl,
+		failOnStatusCode: false
+	}).then(function (response) {
+		if(response.status === 303){
+			const newCount = count + 1;
+			if(newCount < 5){
+				cy.wait(2000)
+				waitUntilRegistryIsReady(registryName, newCount)
```
Comment (janbj) [inline cypress/support/commands.js]: Have you seen this? https://docs.cypress.io/api/commands/request.html#Request-Polling

File: cypress/support/commands.js [hunk 7]
```diff
@@ -23,3 +23,122 @@
...
+	})
+}
+
+// create registry
+function createRegistry(registryName, apiAdminApiKey, metadataFile, createEntity) {
+	cy.log('Using apiKey ' + apiAdminApiKey)
+	cy.fixture(metadataFile)
+	.then(function (testSchema) {
+		testSchema.id = registryName;
+		let createUrl = '/registry';
```
Comment (janbj) [inline cypress/support/commands.js]: Constants in case we change the api paths?

File: cypress/support/commands.js [hunk 8]
```diff
@@ -23,3 +23,122 @@
...
+					cy.createEntity(registryName, registryAdminApiKey, testDataFile)
+				})
+			}
+		})
+	})
+}
+
+// create entity in existing registry
+function createEntity(registryName, apiKey, dataFile) {
+	let entityAddUrl = '/registry/' + registryName + '/entity';
```
Comment (janbj) [inline cypress/support/commands.js]: log when we are calling functions and commands?

File: cypress/support/index.js [hunk 9]
```diff
@@ -21,48 +21,31 @@ import './commands'
 
 beforeEach(function(){
 	let uuid = require('uuid');
```
Comment (janbj) [inline cypress/support/index.js]: create package.json file for dependencies?

File: cypress/support/step_definitions/anonymous_user_is_resource_modified.js [hunk 10]
```diff
@@ -5,15 +5,16 @@
...
 	cy.get('@response').then((response) => {
 		cy.log('response = ' + response.headers)
 		.then(() => {
 			expect(response.headers['date']).to.not.be.undefined
 			expect(response.headers['date']).to.not.equal('')
-			expect(response.headers['last-modified']).to.not.be.undefined
-			expect(response.headers['last-modified']).to.not.equal('')
-			expect(response.headers['etag']).to.not.be.undefined
-			expect(response.headers['etag']).to.not.equal('')
+//			expect(response.headers['last-modified']).to.not.be.undefined
```
Comment (janbj) [inline cypress/support/step_definitions/anonymous_user_is_resource_modified.js]: Say why this is commented out

File: cypress/support/step_definitions/api_admin_delete_data_from_populated_registry.js [hunk 11]
```diff
@@ -24,10 +25,10 @@ when('the API admin user uses the API key and submits a request to delete the da
 then('the API admin user receives information that the data is deleted', () => {
 	cy.get('@registryName').then((registryName) => {
 		cy.get('@deleteConfirmation').then((deleteComfirmation) => {
-			assert.isEqual(deleteComfirmation, 'Registry ' + registryName + ' has been emptied')
+			expect(deleteComfirmation).is.equal('Registry ' + registryName + ' has been emptied')
 			let registryUrl = '/registry' + registryName;
 			cy.request(registryUrl).then((response) => {
-				assert.isEqual(response.body.size, 0);
+//				assert.isEqual(response.body.size, 0);
```
Comment (janbj) [inline cypress/support/step_definitions/api_admin_delete_data_from_populated_registry.js]: Why?

File: lambdaTrustRoleConfig/lambdaTrustRolePolicy.json [hunk 12]
```diff
@@ -15,6 +15,8 @@
         "dynamodb:GetItem",
         "dynamodb:UpdateItem",
         "dynamodb:DeleteItem",
+        "dynamodb:Scan",
+        "dynamodb:ListTables",
```
Comment (janbj) [inline lambdaTrustRoleConfig/lambdaTrustRolePolicy.json]: Delete this file

File: cypress/integration/anonymousUser.feature (hunk 4 above)
Comment (Eiriknil) [reply cypress/integration/anonymousUser.feature]: yes

File: cypress/support/commands.js (hunk 5 above)
Comment (Eiriknil) [reply cypress/support/commands.js]: yes

File: cypress/support/commands.js (hunk 7 above)
Comment (Eiriknil) [reply cypress/support/commands.js]: Need to build urls with registrynames and entityids. Maybe for later

**Pull request merged**